src/dnode/src/dnodeMain.c

```diff
@@ -41,6 +41,25 @@ static void dnodeSetRunStatus(SDnodeRunStatus status);
 static void signal_handler(int32_t signum, siginfo_t *sigInfo, void *context);
 static void dnodeCheckDataDirOpenned(char *dir);
 static SDnodeRunStatus tsDnodeRunStatus = TSDB_DNODE_RUN_STATUS_STOPPED;
+static int32_t dnodeInitSteps();
+static void dnodeCleanupSteps(int32_t stepId);
+
+typedef struct {
+  const char *const name;
+  int               (*init)();
+  void              (*cleanup)();
+} DnodeStep;
```

Comment:
For TDengine convention,  structure name starts with Upper case 'S'. Also, may you change the name to SDnodeModule.  It makes more readable.

Reply:
I name the struct as SDnodeComponent to make it consistent with the init and cleanup function, and also to eliminate confusion since the code already has 'modules'.